pennylane/devices/default_clifford.py

```diff
@@ -466,50 +478,256 @@ def simulate(
+        return op
+
+    # pylint:disable=too-many-return-statements
+    def _convert_op_to_linear_comb(self, meas_obs, coeffs, paulis):
+        """Convert a PennyLane observable to a linear combination of stim Pauli terms"""
+
+        # Case for simple Pauli terms
+        if isinstance(meas_obs, (qml.Identity, qml.PauliZ, qml.PauliX, qml.PauliY)):
+            coeffs.append(1.0)
+            paulis.append((_GATE_OPERATIONS[meas_obs.name], meas_obs.wires))
+            return coeffs, paulis
+
+        # Case for simple Pauli tensor
+        if isinstance(meas_obs, qml.operation.Tensor):
+            coeffs.append(1.0)
+            paulis.append(
+                ("".join([_GATE_OPERATIONS[name] for name in meas_obs.name]), meas_obs.wires)
+            )
+            return coeffs, paulis
+
+        # Case for a Hamiltonian
+        if isinstance(meas_obs, qml.Hamiltonian):
+            cof, obs = meas_obs.terms()
+            coeffs.extend(cof)
+            for ob in obs:
+                expec = (
+                    _GATE_OPERATIONS[ob.name]
+                    if not isinstance(ob.name, list)
+                    else "".join([_GATE_OPERATIONS[o] for o in ob.name])
+                )
+                paulis.append((expec, ob.wires))
+            return coeffs, paulis
+
+        # Case for Sum
+        if isinstance(meas_obs, Sum):
+            for op in meas_obs:
+                coeffs, paulis = self._convert_op_to_linear_comb(op, coeffs, paulis)
+            return coeffs, paulis
+
+        # Case for higher arithmetic depth for prod-type observables
+        if meas_obs.arithmetic_depth > 1 and isinstance(meas_obs, (Prod, SProd)):
+            check_op_math = qml.operation.active_new_opmath()
+            qml.operation.enable_new_opmath()
+            meas_obs_simp = meas_obs.simplify()
+            if not check_op_math:
+                qml.operation.disable_new_opmath()
+
+            # Recurse only if the simplification happened
+            if meas_obs_simp != meas_obs:
+                coeffs, paulis = self._convert_op_to_linear_comb(meas_obs_simp, coeffs, paulis)
+                return coeffs, paulis
+
+        # Case for Prod
+        if isinstance(meas_obs, Prod):
+            coeffs.append(1.0)
+            paulis.append(
+                (
+                    "".join([_GATE_OPERATIONS[op.name] for op in meas_obs.terms()[1][0]]),
+                    meas_obs.wires,
+                )
+            )
+            return coeffs, paulis
+
+        # Case for SProd
+        if isinstance(meas_obs, SProd):
+            cof, obs = meas_obs.terms()
+            gate_ops = obs
+            if len(gate_ops) == 1 and isinstance(gate_ops[0], Prod):
+                gate_ops = obs[0].terms()[1][0]
+            coeffs.extend(cof)
+            paulis.append(("".join([_GATE_OPERATIONS[op.name] for op in gate_ops]), meas_obs.wires))
+            return coeffs, paulis
+
+        # Add support for more case when the time is right
```

Comment:
please make sure to add a story to track any future work that is needed/nice to have:)